pennylane/templates/subroutines/flip_sign.py

```diff
@@ -0,0 +1,191 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the FlipSign template.
+"""
+
+import numpy as np
+import pennylane as qml
+from pennylane.operation import Operation, AnyWires
+
+
+class FlipSign(Operation):
+    r"""FlipSign operator flips the sign for a given basic state.
+
+    In a nutshell, this class perform the following operation:
+
+    FlipSign(n):math:`|m\rangle = -|m\rangle` if m = n
+    FlipSign(n):math:`|m\rangle = |m\rangle` if m != n
+
+    Where n is the basic state to flip and m is the input.
+    It flips the sign of the state.
+
+    Args:
+        n (array[int] or int): binary array or integer value representing the state to flip the sign
+        wires (array[int]): number of wires that the operator acts on
+
+    Raises:
+        ValueError: "expected an integer array for wires "
+        ValueError: "expected at least one wire representing the qubit "
+        ValueError: "expected an integer binary array or integer number for basic flipping state "
+        ValueError: "expected an integer equal or greater than zero for basic flipping state "
+
+    .. seealso:: :func:`~.relevant_func`, :class:`~.RelevantClass` (optional)
+
+    .. details::
+
+        :title: Usage Details
+
+        The template is used inside a qnode.
+        The number of shots has to be explicitly set on the device when using sample-based measurements:
+
+        .. code-block:: python
+
+            dev = qml.device("default.qubit", wires=2, shots = 1)
+
+            @qml.qnode(dev)
+            def circuit():
+               for wire in list(range(2)):
+                    qml.Hadamard(wires = wire)
+               qml.FlipSign([1,0], wires = list(range(2)))
+               return qml.sample()
+
+            circuit()
+
+        The result for the above circuit is:
+
+            >>> print(drawer())
+            0: ──H─╭FlipSign──H─┤  Sample
+            1: ──H─├FlipSign──H─┤  Sample
+            2: ──H─├FlipSign──H─┤  Sample
+            3: ──H─├FlipSign──H─┤  Sample
+            4: ──H─╰FlipSign──H─┤  Sample
+
+    """
+
+    num_wires = AnyWires
+
+    def __init__(self, n, wires, do_queue=True, id=None):
+
+        if not isinstance(wires, list):
+            raise ValueError("expected an integer array for wires ")
+
+        if len(wires) == 0:
+            raise ValueError("expected at least one wire representing the qubit ")
+
+        if np.array(wires).dtype != np.dtype("int"):
+            raise ValueError("expected an integer array for wires")
+
+        if isinstance(n, list):
```

Comment:
it is a problem to ask if it is a list. Because you might get an array or a tensor too. What I would do is say, if `isinstance(n, int)`  do that,  else do this other thing.

Reply:
code modified!